google-cloud-util/google-cloud-compat-checker/src/main/java/com/google/cloud/compatchecker/GoogleCloudCompatChecker.java

```diff
@@ -43,30 +44,96 @@ protected void logProperty(String s, String s1) {
   }
 
   public static void main(String[] args) {
+    Boolean checkResult = check();
+    if (checkResult != null && !checkResult) {
+      System.exit(1);
+    }
+  }
+
+  public static Boolean check() {
```

Comment:
Please split this method. Potentially grouping OS, JVM and OpenSSL things (potentially changing output order too: first OS details + check, and only then JVM details + check etc). Codacy-bot complains that it this method is too complex and it is right. Also, splitting makes adding more stuff easier (and probably there will be more stuff).